packages/core/src/browser/tree/tree-widget.tsx

```diff
@@ -850,18 +931,34 @@ export class TreeWidget extends ReactWidget implements StatefulWidget {
         return { paddingLeft };
     }
 
-    protected getPaddingLeft(node: TreeNode, props: NodeProps): number {
-        return props.depth * this.props.leftPadding + (this.needsExpansionTogglePadding(node) ? this.props.expansionTogglePadding : 0);
+    /**
+     * Add right padding (3px) for icons located in the root level of the tree during rendering.
+     * @param node the tree node.
+     * @param props the node properties.
+     */
+    protected needsRootLevelIconPadding(node: TreeNode, props: NodeProps): boolean {
+        return (props.depth === 0 && !this.isExpandable(node));
     }
 
     /**
+     * Code is kept here to prevent broken api
      * If the node is a composite, a toggle will be rendered.
      * Otherwise we need to add the width and the left, right padding => 18px
+     * @param node the tree node.
      */
     protected needsExpansionTogglePadding(node: TreeNode): boolean {
         return !this.isExpandable(node);
     }
 
+    /**
+     * Return the left padding for a tree node.
+     * @param node the tree node.
+     * @param props the node properties.
+     */
+    protected getPaddingLeft(node: TreeNode, props: NodeProps): number {
+        return this.props.leftPadding + (this.needsRootLevelIconPadding(node, props) ? this.props.rootLevelIconPadding : 0);
```

Comment:
why `needsExpansionTogglePadding` is not used anymore? It breaks all trees outside of this repo. We should make sure that all existing methods are respected before. TreeWidget is used and customized a lot.